readthedocs/rtd_tests/tests/test_config_integration.py

```diff
@@ -199,23 +211,582 @@ def test_conda(self, load_config):
+
+    @patch('readthedocs.projects.models.Project.repo_nonblockinglock', new=MagicMock())
+    @patch('readthedocs.projects.tasks.UpdateDocsTaskStep.build_docs_class')
+    @patch('readthedocs.doc_builder.backends.sphinx.SearchBuilder.build')
+    @patch('readthedocs.doc_builder.backends.sphinx.HtmlBuilder.build')
+    @patch('readthedocs.doc_builder.backends.sphinx.HtmlBuilder.append_conf')
+    def test_build_formats_only_pdf(
+            self, append_conf, html_build, search_build, build_docs_class,
+            checkout_path, tmpdir):
+        """
+        Only the pdf format is build.
+        """
+        checkout_path.return_value = str(tmpdir)
+        self.create_config_file(tmpdir, {'formats': ['pdf']})
+
+        update_docs = self.get_update_docs_task()
+        outcomes = update_docs.build_docs()
+
+        # Only pdf extra format was triggered
+        assert outcomes['html']
+        assert outcomes['search']
+        build_docs_class.assert_called_with('sphinx_pdf')
+        assert outcomes['pdf']
+        assert not outcomes['localmedia']
+        assert not outcomes['epub']
+
+    @patch('readthedocs.projects.tasks.UpdateDocsTaskStep.update_documentation_type', new=MagicMock())
+    @patch('readthedocs.projects.tasks.UpdateDocsTaskStep.setup_python_environment', new=MagicMock())
+    @patch('readthedocs.projects.tasks.UpdateDocsTaskStep.build_docs', new=MagicMock())
+    @patch('readthedocs.doc_builder.environments.BuildEnvironment.failed', new_callable=PropertyMock)
+    def test_conda_environment(self, build_failed, checkout_path, tmpdir):
+        build_failed.return_value = False
+        checkout_path.return_value = str(tmpdir)
+        conda_file = 'environmemt.yml'
+        apply_fs(tmpdir, {conda_file: ''})
+        base_path = self.create_config_file(
+            tmpdir,
+            {
+                'conda': {'environment': conda_file}
+            }
+        )
+
+        update_docs = self.get_update_docs_task()
+        update_docs.run_build(docker=False, record=False)
+
+        conda_file = path.join(str(base_path), conda_file)
+        assert update_docs.config.conda.environment == conda_file
+        assert isinstance(update_docs.python_env, Conda)
+
+    def test_default_build_image(self, checkout_path, tmpdir):
+        checkout_path.return_value = str(tmpdir)
+        build_image = 'readthedocs/build:latest'
+        self.create_config_file(tmpdir, {})
+        update_docs = self.get_update_docs_task()
+        assert update_docs.config.build.image == build_image
+
+    def test_build_image(self, checkout_path, tmpdir):
+        checkout_path.return_value = str(tmpdir)
+        build_image = 'readthedocs/build:stable'
+        self.create_config_file(
+            tmpdir,
+            {'build': {'image': 'stable'}},
+        )
+        update_docs = self.get_update_docs_task()
+        assert update_docs.config.build.image == build_image
+
+    def test_custom_build_image(self, checkout_path, tmpdir):
```

Comment:
What would happen if they define a build image in the YAML and _also_ there is a `project.container_image`?